ui/src/main/kotlin/edu/artic/ui/BaseFragment.kt

```diff
@@ -37,21 +44,38 @@ abstract class BaseFragment : Fragment() {
 
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
         return inflater.inflate(layoutResId, container, false)
-
     }
 
+    protected open fun hasTransparentStatusBar(): Boolean = false
+
+    protected open fun hasHomeAsUpEnabled(): Boolean = true
+
     private fun updateToolbar(view: View) {
         toolbar = view.findViewById(R.id.toolbar)
         if (toolbar != null) {
             baseActivity.setSupportActionBar(toolbar)
             baseActivity.supportActionBar?.apply {
                 setDisplayShowTitleEnabled(true)
-                setDisplayHomeAsUpEnabled(true)
+                setDisplayHomeAsUpEnabled(hasHomeAsUpEnabled())
                 setDisplayShowHomeEnabled(true)
             }
 
             baseActivity.title = title
         }
+
+        collapsingToolbar = view.findViewById(R.id.collapsingToolbar)
+        collapsingToolbar?.let {
```

Comment:
switch with `apply()` if you want. usually I use it for blocks of single-object manipulation.